src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisConnectorService.java

```diff
@@ -97,15 +110,65 @@ public void executePipeline(String feature, String variant, Object executionDTO)
      * @param variant      The variant of the feature to execute
      * @param executionDTO The DTO sent as a body for the execution
      */
-    public void executeLectureWebhook(String variant, PyrisWebhookLectureIngestionExecutionDTO executionDTO) {
+    public void executeLectureAddtionWebhook(String variant, PyrisWebhookLectureIngestionExecutionDTO executionDTO) {
         var endpoint = "/api/v1/webhooks/lectures/" + variant;
         try {
             restTemplate.postForEntity(pyrisUrl + endpoint, objectMapper.valueToTree(executionDTO), Void.class);
         }
+        catch (HttpStatusCodeException e) {
+            log.error("Failed to send lecture unit {} to Pyris: {}", executionDTO.pyrisLectureUnit().lectureUnitId(), e.getMessage());
+            throw toIrisException(e);
+        }
+        catch (RestClientException | IllegalArgumentException e) {
+            log.error("Failed to send lecture unit {} to Pyris: {}", executionDTO.pyrisLectureUnit().lectureUnitId(), e.getMessage());
+            throw new PyrisConnectorException("Could not fetch response from Pyris");
+        }
+    }
+
+    /**
+     * uses send an api call to get the ingestion state in Pyris
+     *
+     * @param courseId      id of the course
+     * @param lectureId     id of the lecture
+     * @param lectureUnitId id of the lectureUnit to check in the Pyris vector database
+     * @return The ingestion state of the lecture Unit
+     *
+     */
+    IngestionState getLectureUnitIngestionState(long courseId, long lectureId, long lectureUnitId) {
+        try {
+            String encodedBaseUrl = URLEncoder.encode(artemisBaseUrl, StandardCharsets.UTF_8);
+            String url = String.format("%s/api/v1/courses/%d/lectures/%d/lectureUnits/%d/ingestion-state?base_url=%s", pyrisUrl, courseId, lectureId, lectureUnitId,
+                    encodedBaseUrl);
+            Map<String, String> response = restTemplate.getForObject(url, Map.class);
+            String stateString = response.get("state");
+            IngestionState state = IngestionState.valueOf(stateString);
+            if (state != IngestionState.DONE) {
+                if (pyrisJobService.currentJobs().stream().filter(job -> job instanceof IngestionWebhookJob).map(job -> (IngestionWebhookJob) job)
+                        .anyMatch(ingestionJob -> ingestionJob.courseId() == courseId && ingestionJob.lectureId() == lectureId && ingestionJob.lectureUnitId() == lectureUnitId)) {
+                    return IngestionState.IN_PROGRESS;
+                }
+            }
+            return state;
+        }
+        catch (RestClientException e) {
+            log.error("Error fetching ingestion state for lecture {}, lecture unit {}, baseUrl {}", lectureId, lectureUnitId, pyrisUrl, e);
+            return IngestionState.ERROR;
```

Comment:
What kind of errors can occur here? Would it make sense to let the caller know what kind of error occurred?

Reply:
For the client we only return ingestion error because it can be part of one call to ingest multiple lecture units, where some can succeed while others may fail. That's why I only return the IngestionState. I added a more specific statement for the error on the server side.